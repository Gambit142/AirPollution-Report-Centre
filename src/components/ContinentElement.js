import { useSelector } from 'react-redux';
import { useState } from 'react';
import { Link } from 'react-router-dom';
import { FiSettings } from 'react-icons/fi';
import logo from '../assets/Africa.png';

const ContinentElement = () => {
  const [searchCountry, setSearchCountry] = useState('');
  const { continentReducer } = useSelector((state) => state);
  if (!continentReducer) return <h1>Loading</h1>;
  const filteredCountries = continentReducer.filter((c) => c.country.includes(searchCountry));
  const totalPopulation = !continentReducer ? 0 : continentReducer.reduce((total, item) => ({
    population: total.population + item.population,
  }), { population: 0 });

  return (
    <>
      <nav className="continent-navbar">
        <span>2021</span>
        <span>Continent</span>
        <FiSettings className="settings-icon" />
      </nav>
      <div className="headline">
        <div>
          <img className="africa" src={logo} alt="Map of Africa" />
          <div className="african-details">
            <span data-testid="map-item-continent">Africa</span>
            <span>
              {`${continentReducer.length}
               Countries`}
            </span>
          </div>
        </div>
      </div>
      <div className="total-population">{`Total Population: ${totalPopulation.population.toLocaleString()}`}</div>
      <input
        type="text"
        name="search-countries"
        id="search-countires"
        placeholder="Search Country..."
        value={searchCountry}
        onChange={(e) => setSearchCountry(e.target.value)}
      />
      <section className="country-container">
        {filteredCountries.map(({
          updated,
          countryInfo: {
            _id: id, flag, lat, long,
          }, country, population,
        }) => (
          <div key={id} className="country-div">
            <Link
              to={{
                pathname: `/country/${country}`,
                state: {
                  long, lat, country, flag,
                },
              }}
            >
              <div className="date">{(new Date(updated)).toDateString()}</div>
              <div className="country-details">
                <img className="flag" src={flag} alt={`${country}'s Flag`} />
                <div>
                  <h2 className="country-name">{country}</h2>
                  <h2 className="population">{population.toLocaleString()}</h2>
                </div>
              </div>
            </Link>
          </div>
        ))}
      </section>
    </>
  );
};

export default ContinentElement;
