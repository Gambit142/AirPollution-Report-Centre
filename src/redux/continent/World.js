import Africa from '../../assets/Africa.png';
import Europe from '../../assets/europe.png';
import Asia from '../../assets/asia.png';
import SouthAmerica from '../../assets/southamerica.png';
import NorthAmerica from '../../assets/northamerica.png';
import Australia from '../../assets/australia2.png';

const initialState = [
  {
    name: 'Africa',
    image: Africa,
  },
  {
    name: 'Europe',
    image: Europe,
  },
  {
    name: 'Asia',
    image: Asia,
  },
  {
    name: 'South America',
    image: SouthAmerica,
  },
  {
    name: 'North America',
    image: NorthAmerica,
  },
  {
    name: 'Australia-Oceania',
    image: Australia,
  },
];

const worldReducer = (state = initialState) => state;

export default worldReducer;
