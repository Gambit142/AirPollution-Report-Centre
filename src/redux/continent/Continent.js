import { fetchContinentApi } from '../../components/fetchApi';

export const FETCH_CONTINENT = 'airpollution/continent/FETCH_CONTINENT';
const initialState = [];

export const fetchContinent = () => async (dispatch) => {
  const continentData = await fetchContinentApi();
  dispatch({
    type: FETCH_CONTINENT,
    continentData,
  });
};

const continentReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_CONTINENT:
      return action.continentData;
    default:
      return state;
  }
};

export default continentReducer;
