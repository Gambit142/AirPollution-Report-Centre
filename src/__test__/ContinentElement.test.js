import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import store from '../redux/ConfigureStore';
import ContinentElement from '../components/ContinentElement';
import { FETCH_CONTINENT } from '../redux/continent/Continent';

store.dispatch({
  type: FETCH_CONTINENT,
  continentData: [
    {
      updated: 1637328762088,
      country: 'Mozambique',
      population: 40135620,
      countryInfo: {
        _id: 4,
        lat: 33,
        long: 65,
        flag: 'https://disease.sh/assets/img/flags/af.png',
      },
    },
  ],
});
describe('Check All tests for CountryElement Component', () => {
  test('If it renders content on the page', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <ContinentElement />
        </Router>
      </Provider>,
    );
    expect(getByTestId('map-item-continent')).toBeInTheDocument();
    expect(screen.getByText('Mozambique')).toBeInTheDocument();
  });
  test('If it creates exact snapsshot of component', async () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <ContinentElement />
        </Router>
      </Provider>,
    );
    screen.debug();
    expect(tree).toMatchSnapshot();
  });
});
