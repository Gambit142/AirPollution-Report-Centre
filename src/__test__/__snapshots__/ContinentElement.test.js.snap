// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check All tests for CountryElement Component If it creates exact snapsshot of component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="continent-navbar"
      >
        <span>
          2021
        </span>
        <span>
          Continent
        </span>
        <svg
          class="settings-icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </nav>
      <div
        class="headline"
      >
        <div>
          <img
            alt="Map of Africa"
            class="africa"
            src="Africa.png"
          />
          <div
            class="african-details"
          >
            <span
              data-testid="map-item-continent"
            >
              Africa
            </span>
            <span>
              1
               Countries
            </span>
          </div>
        </div>
      </div>
      <div
        class="total-population"
      >
        Total Population: 40,135,620
      </div>
      <input
        id="search-countries"
        name="search-countries"
        placeholder="Search Country..."
        type="text"
        value=""
      />
      <section
        class="country-container"
      >
        <div
          class="country-div"
        >
          <a
            href="/country/Mozambique"
          >
            <div
              class="date"
            >
              Fri Nov 19 2021
            </div>
            <div
              class="country-details"
            >
              <img
                alt="Mozambique's Flag"
                class="flag"
                src="https://disease.sh/assets/img/flags/af.png"
              />
              <div>
                <h2
                  class="country-name"
                >
                  Mozambique
                </h2>
                <h2
                  class="population"
                >
                  40,135,620
                </h2>
              </div>
            </div>
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <nav
      class="continent-navbar"
    >
      <span>
        2021
      </span>
      <span>
        Continent
      </span>
      <svg
        class="settings-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
    </nav>
    <div
      class="headline"
    >
      <div>
        <img
          alt="Map of Africa"
          class="africa"
          src="Africa.png"
        />
        <div
          class="african-details"
        >
          <span
            data-testid="map-item-continent"
          >
            Africa
          </span>
          <span>
            1
               Countries
          </span>
        </div>
      </div>
    </div>
    <div
      class="total-population"
    >
      Total Population: 40,135,620
    </div>
    <input
      id="search-countries"
      name="search-countries"
      placeholder="Search Country..."
      type="text"
      value=""
    />
    <section
      class="country-container"
    >
      <div
        class="country-div"
      >
        <a
          href="/country/Mozambique"
        >
          <div
            class="date"
          >
            Fri Nov 19 2021
          </div>
          <div
            class="country-details"
          >
            <img
              alt="Mozambique's Flag"
              class="flag"
              src="https://disease.sh/assets/img/flags/af.png"
            />
            <div>
              <h2
                class="country-name"
              >
                Mozambique
              </h2>
              <h2
                class="population"
              >
                40,135,620
              </h2>
            </div>
          </div>
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
